### ISSUE TITLE
[FEATURE] Branch protection rules for `mt8195/*`

### ISSUE THREAD

**Issue by afq984** [thesofproject/sof#5457]
**Is your feature request related to a problem? Please describe.**
Prevent accidentally overwriting code in `mt8195/*` branches which are used to build firmware releases.
Currently the only branch that matches this pattern is `mt8195/v0.4`.

**Describe the solution you'd like**
Set branch protection rules on `mt8195/*`.

**Describe alternatives you've considered**
Be careful to not use `git push --force ...`

**Comment by lgirdwood** [comment]
@afq984 who should have Write access ?

**Comment by dbaluta** [comment]
@lgirdwood since we are here could also give write access for imx-stable-v2.0 only to @dbaluta @paulstelian97 and @TangleZ

**Comment by lgirdwood** [comment]
@dbaluta Sure. 
@balakishorepati @bhiregou any similar request ?

**Comment by lgirdwood** [comment]
@dbaluta @afq984 done - please try.

**Comment by cujomalainey** [comment]
@afq984 are we good to close this?

**Comment by afq984** [comment]
I'll let @dbaluta close this

**Comment by afq984** [comment]
@lgirdwood ah, can the protection rules be lifted to just "protection" only? i.e. drop all extra protection such as signed commits, PR, status checks, just prevent force push & deletion

**Comment by afq984** [comment]
Works for me, thanks again!

**Comment by bhiregoudar** [comment]
> @dbaluta Sure. @balakishorepati @bhiregou any similar request ?

@lgirdwood : We come back tomorrow on this.

**Comment by bhiregoudar** [comment]
> > @dbaluta Sure. @balakishorepati @bhiregou any similar request ?
> 
> @lgirdwood : We come back tomorrow on this.

Could you help add @bhiregoudar @balakishorepati @sunilkumardommati @maruthi-machani for amd-rn-stable branch

**Comment by lgirdwood** [comment]
> > > @dbaluta Sure. @balakishorepati @bhiregou any similar request ?
> > 
> > 
> > @lgirdwood : We come back tomorrow on this.
> 
> Could you help add @bhiregoudar @balakishorepati @sunilkumardommati @maruthi-machani for amd-rn-stable branch

Done, but I can't add @maruthi-machani yet until GH invite is accepted. @maruthi-machani please let me know when you accept the invite.

**Comment by dbaluta** [comment]
All good for imx-stable-v2.0 branch. @lgirdwood can this rules be inherited for future imx-stable-* branches as @cujomalainey suggested?

**Comment by lgirdwood** [comment]
> All good for imx-stable-v2.0 branch. @lgirdwood can this rules be inherited for future imx-stable-* branches as @cujomalainey suggested?

Yes, I've made the wildcard "imx-*" so rules should apply to all IMX branches.

**Comment by lgirdwood** [comment]
Good to close ?

**Comment by maruthi-machani** [comment]
> > > > @dbaluta Sure. @balakishorepati @bhiregou any similar request ?
> > > 
> > > 
> > > @lgirdwood : We come back tomorrow on this.
> > 
> > 
> > Could you help add @bhiregoudar @balakishorepati @sunilkumardommati @maruthi-machani for amd-rn-stable branch
> 
> Done, but I can't add @maruthi-machani yet until GH invite is accepted. @maruthi-machani please let me know when you accept the invite.

Accepted the requested.
